test.js

```diff
@@ -99,21 +99,18 @@ async function test() {
             switch (type_id) {
             case duckdb_native.duckdb_type.DUCKDB_TYPE_BIGINT: {
                 // seems ugly as hell but should be jit-ted away
-                var typed_validity_buf;
                 const validity_buf =
                     duckdb_native.copy_buffer(duckdb_native.duckdb_vector_get_validity(vector),
                                               Math.ceil(n / 64) * 8); // this will be null if all rows are valid
-                if (validity_buf != null) {
-                    typed_validity_buf = new BigUint64Array(validity_buf.buffer);
-                }
+                const typed_validity_buf = validity_buf ? new BigUint64Array(validity_buf.buffer) : null;
 
                 const data_buf = duckdb_native.copy_buffer(duckdb_native.duckdb_vector_get_data(vector), 8 * n);
-                const typed_data_arr = new BigInt64Array(data_buf.buffer);
+                const typed_data_arr = data_buf ? new BigInt64Array(data_buf.buffer) : null;
                 const vector_data = new Array(n)
                 for (let row_idx = 0; row_idx < n; row_idx++) {
                     // interpret validity mask bytes
-                    const row_is_valid = validity_buf == null || (typed_validity_buf[Math.floor(row_idx / 64)] &
-                                                                  (1n << BigInt(row_idx % 64))) > 0;
+                    const row_is_valid = typed_validity_buf == null || (typed_validity_buf[Math.floor(row_idx / 64)] &
+                                                                  (BigInt(1) << BigInt(row_idx % 64))) > 0;
```

Comment:
Not all JS targets can handle the (newer) `bigint` syntax, so I'm just using the older one consistently to avoid issues.